File: Aufgabe_3/AdminShellApp/src/ShellManager/Util/Time.java
package ShellManager.Util;

import com.google.gson.Gson;
import com.mashape.unirest.http.HttpResponse;
import com.mashape.unirest.http.Unirest;
import com.mashape.unirest.http.exceptions.UnirestException;
import java.util.Calendar;
import org.json.JSONException;

public class Time {

    private final String timeServerURI = "http://localhost/time/index.php";
    private long timeInMills;

    public Time() {
            this.timeInMills = getLocalTime();
    }

    public long getCurrentServerTime() throws UnirestException, JSONException {
            TimeDTO t = getCurrentServerTimeAsJSON();
            return t.getTime();
    }
    
    private TimeDTO getCurrentServerTimeAsJSON() throws UnirestException{
        HttpResponse<String> response = Unirest.get(timeServerURI).asString();
        
        Gson g = new Gson();
        TimeDTO t = g.fromJson(response.getBody(), TimeDTO.class);
        
        return t;
    }

    public long getTimeInSecPlus30Min() {

            return timeInMills+(30*60);
    }

    public long getTimeInSec() {

            return timeInMills;
    }
    
    private long getLocalTime(){
        Calendar c = Calendar.getInstance();
      
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH)+1;
        int day  = c.get(Calendar.DAY_OF_MONTH);
        int hour = c.get(Calendar.HOUR_OF_DAY);
        int min  = c.get(Calendar.MINUTE);
        long sec = 60L;
        long result = 0L;
        
        result += min * sec;
        result += hour * (sec * sec);
        result += day  * (24 * sec * sec);
        result += month* (30 * 24 * sec * sec);
        result += year * (12 * 30 * 24 * sec * sec);
        
        return result;
    }
    
}

class TimeDTO {
    
    private long time;
    private long year;
    private long month;
    private long day;
    private long hour;
    private long min;

    public TimeDTO(long time, long year, long month, long day, long hour, long min) {
        this.time = time;
        this.year = year;
        this.month = month;
        this.day = day;
        this.hour = hour;
        this.min = min;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public long getYear() {
        return year;
    }

    public void setYear(long year) {
        this.year = year;
    }

    public long getMonth() {
        return month;
    }

    public void setMonth(long month) {
        this.month = month;
    }

    public long getDay() {
        return day;
    }

    public void setDay(long day) {
        this.day = day;
    }

    public long getHour() {
        return hour;
    }

    public void setHour(long hour) {
        this.hour = hour;
    }

    public long getMin() {
        return min;
    }

    public void setMin(long min) {
        this.min = min;
    }

       
    
    
}

File: Aufgabe_3/PHP_WebApp/Util/Time.php
<?php

/**
 * Created by PhpStorm.
 * User: dima
 * Date: 11.05.16
 * Time: 11:23
 */
class Time
{

    protected $timeInSec;

    public function __construct()
    {
        $this -> year   = date("Y");
        $this -> month  = date("n");
        $this -> day    = date("j");

        $this -> hour = date("H");
        $this -> min  = date("i");


        $this -> timeInSec =
            ($this->min  * 60)+
            ($this->hour * 60 * 60) +
            ($this->day  * 24 * 60 * 60) +
            ($this->month * 30 * 24 * 60 * 60) +
            ($this->year * 12 * 30 * 24 * 60 * 60);
    }


    public function getTimeInSec()
    {
        return $this->timeInSec;
    }

    public function getTimeInSecPlus30min()
    {
        $minInSec30 = 30*60;
        return $this->timeInSec + $minInSec30;
    }

    /**
     * @return bool|string
     */
    public function getDay()
    {
        return $this->day;
    }

    /**
     * @return bool|string
     */
    public function getHour()
    {
        return $this->hour;
    }

    /**
     * @return bool|string
     */
    public function getMin()
    {
        return $this->min;
    }

    /**
     * @return bool|string
     */
    public function getMonth()
    {
        return $this->month;
    }

    /**
     * @return bool|string
     */
    public function getYear()
    {
        return $this->year;
    }



}

?>